### DISCUSSION TITLE
Questions Regarding Stream Management and Timeout Configuration in Ant Media

### DISCUSSION THREAD

**Discussion by ankit-akteena** [ant-media/Ant-Media-Server#6853] [Q&A] [answered]
We are currently exploring Ant Media for our streaming needs and have a few questions we couldn’t find answers to in the documentation or GitHub discussions:

1. **Single-use Stream**: Is there a way to create a stream that can only be published once? In our use case, the stream should be published only once, and after it ends, a webhook should be triggered to deactivate the stream. Is there a way to achieve this?

2. **Stream Timeout Configuration**: When a stream is published from our source, it starts broadcasting on Ant Media. However, in case of a network disruption on the source side, the stream stops automatically after about 40-60 seconds. Is this timeout configurable?

**Comment by rahul78275** [comment]
Hello @ankit-akteena 

**1. Single-use Stream**: You can achieve this from your application side. Before publishing the streaming you can check this on your db that this stream exist or not, If stream published earlier you can either append something to streamid OR display any error.

Here is the [documentation](https://antmedia.io/docs/guides/advanced-usage/webhooks/) for webhook. You can find **liveStreamEnded** event.

**Reply by ankit-akteena** [reply]
Yes this can be done , but for my use case the source cannot check the application in real-time because of this I was wondering if antmeadia had a way to stop the stream after a single publish.

**Reply by Mohit-3196** [reply, marked as answer]
Thank you @rahul78275 :)

I also have a workaround solution for the single-se stream @ankit-akteena 
- [Disable Accepting undefined streams](https://antmedia.io/docs/guides/stream-security/accepting-undefined-streams/) 
- Create a stream using the [createBroadcast API](https://antmedia.io/rest/#/default/createBroadcast).
- After the stream is ended, you will get the liveStreamEnded webhook call. Once it is received, call the [deleteBroadcast API](https://antmedia.io/rest/#/default/deleteBroadcast)

This will ensure that the stream is deleted and used only once. Even if someone tries to broadcast the stream again, it will not be accepted by the server. 

2. If you publish with RTMP or SRT, you should enable the retry on the encoder side because it's a push to the server. For the pull streams, the server keeps retrying so there won't be a timeout. 
In the case of WebRTC streams, you can change the timeout by modifying the default timeout value as shown [here](https://antmedia.io/javadoc/io/antmedia/AppSettings.html#webRTCClientStartTimeoutMs)

I hope it helps @ankit-akteena 
Let me know if it doesn't help.

--
Cheers,
Mohit

**Reply by ankit-akteena** [reply]
Thanks for the help @Mohit-3196

**Reply by Mohit-3196** [reply]
My pleasure @ankit-akteena :)

**Reply by ankit-akteena** [reply]
@Mohit-3196  I have a question regarding the **publishTimeoutError** webhook. Despite exploring the available list of webhooks, I haven't encountered this specific webhook while testing streams on my community edition. I am using RTMP via OBS for testing and have tried scenarios like disconnecting the network, but I only receive the **liveStreamEnded** webhook instead of **publishTimeoutError**. Could you help me understand the conditions under which this webhook is triggered and how I can simulate it on my server?

**Reply by Mohit-3196** [reply]
@ankit-akteena,
The **publishTimeoutError** is for webrtc publish & since the community edition does not support webrtc, you do not encounter the error.

**Reply by ankit-akteena** [reply]
Got it , Thanks @Mohit-3196
